dataflow/gemma/README.md

```diff
@@ -0,0 +1,152 @@
+# RunInference on Dataflow Streaming with Gemma
+
+Gemma is a family of lightweight, state-of-the art open models built from research and technology used to create the Gemini models.
+You can use Gemma models in your Apache Beam inference pipelines with the `RunInference` transform.
+
+This example demonstrates how to utilize a Gemma model in a streaming Dataflow pipeline using Pub/Sub sources and sinks.
+
+For more information about using RunInference, see [Get started with AI/ML pipelines](https://beam.apache.org/documentation/ml/overview/) in the Apache Beam documentation.
+
+## Before you begin
+
+Follow the steps in this section to create the necessary environment to run this workflow.
+
+### Enable Google Cloud services
+
+This workflow uses multiple Google Cloud Platform products, including Dataflow, Pub/Sub, Google Cloud Storage, and Artifact Registry. Before you start the workflow, create a Google Cloud project that has the following services enabled:
+
+* Dataflow
+* Pub/Sub
+* Google Cloud Storage
+* Artifact Registry
+
+Using these services incurs billing charges.
+
+You will also need to ensure that your GCP project has Nvidia L4 GPU quota enabled, see [Google Cloud Platform documentation](https://cloud.google.com/compute/resource-usage#gpu_quota) for more information.
+
+### Create a custom container
+
+You need to use Docker to build a custom container. This repository contains a Dockerfile that you can use to build your custom container. Follow the [Google Cloud documentation](https://cloud.google.com/dataflow/docs/guides/build-container-image#build_and_push_the_image) that explains how to build and push a container to Artifact Registry by using Docker.
+
+### Create Pub/Sub topics for input and output
+
+Follow the [Google Cloud documentation for creating Pub/Sub topics](https://cloud.google.com/pubsub/docs/create-topic#pubsub_create_topic-Console). This example uses two: one input topic and one output topic. For input, you will need to have a subscription to the topic to pass to the model. 
+
+### Download and save the model
+
+Save a version of the Gemma 2B model. Downloaded the model from [Kaggle](https://www.kaggle.com/models/keras/gemma/frameworks/keras/variations/gemma_2b_en), and then rename the downloaded archive to `gemma_2B`.
```

Comment:
Maybe a note that this becomes a folder?

Reply:
Done